modules/machine-api-overview.adoc

```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+// * architecture/architecture.adoc
+
+[id='machine-api-overview-{context}']
+= Machine API overview
+
+On fully-managed {product-title} clusters, the Machine API performs all node
+management actions after the cluster installation finishes. Because of this
+system, {product-title} version 4.0 offers an elastic, dynamic provisioning
+method.
+
+The Machine API is a combination of primary resources that are based on the upstream
+link:https://github.com/kubernetes-sigs/cluster-api[Cluster API] project and
+custom {product-title} resources.
+
+The three primary resources are:
+
+`Machines`:: A fundamental unit that describes a node host. A `machine` has a
+class, which describes the types of compute nodes that are offered for different
+cloud platforms. For example, a `machine` type for a worker node on Amazon Web
+Services (AWS) might define a specific machine type and required metadata.
+`MachineClasses`:: A unit that defines a class of `machines` and facilitates
+configuration reuse across `machines` of the same class. This unit functions
+like a `StorageClass` for PersistentVolumeClaims.
+`MachineSets`:: Groups of machines. `MachineSets` are to `machines` as
+`ReplicaSets` are to `Pods`. If you need more `machines` or need to scale them down,
```

Comment:
Actually, this is fine, since [`MachineDeployments` are a separate thing][1].

[1]: https://github.com/openshift/openshift-docs/pull/12880#discussion_r245325123